static/app/utils/profiling/flamegraph/flamegraphStateProvider/flamegraphProfiles.tsx

```diff
@@ -23,13 +27,18 @@ export function flamegraphProfilesReducer(
 ): FlamegraphProfilesState {
   switch (action.type) {
     case 'set selected node': {
-      return {...state, selectedNode: action.payload};
+      return {
+        ...state,
+        selectedNode: action.payload,
+        selectedNodeId: action.payload ? getFlamegraphFrameId(action.payload!) : null,
```

Comment:
@eliashussary why do we need to store these two separately? Wouldn't it make sense to just read the id off the selectedNode?